internal/transport/transport_test.go

```diff
@@ -2746,3 +2746,81 @@ func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
 		t.Errorf("Context timed out")
 	}
 }
+
+// Test that in the event of a graceful client transport shutdown
+// , i.e., clientTransport.Close(), if the conn hung for
```

Comment:
suggestion: , i.e., clientTransport.Close(), if the GOAWAY write is not finished within specified time due to network hang, client should still close without waiting for too long